code/parachain/frame/assets-registry/src/lib.rs

```diff
@@ -117,6 +134,17 @@ pub mod pallet {
 	#[pallet::getter(fn asset_ratio)]
 	pub type AssetRatio<T: Config> = StorageMap<_, Twox128, T::LocalAssetId, Rational, OptionQuery>;
 
+	/// Metadata of an asset
+	#[pallet::storage]
+	#[pallet::getter(fn decimals)]
+	pub type Metadata<T: Config> = StorageMap<
+		_,
+		Twox128,
+		T::LocalAssetId,
+		AssetMetadata<BoundedVec<u8, T::AssetNameMaxChars>, BoundedVec<u8, T::AssetSymbolMaxChars>>,
+		OptionQuery,
+	>;
+
```

Comment:
REVIEW POINT:
Removal of the `ForeignMetadata` storage in favor of a `Metadata` type used by both local and foreign assets